sentry-logback/src/main/java/io/sentry/logback/SentryAppender.java

```diff
@@ -91,8 +90,11 @@ protected void append(@NotNull ILoggingEvent eventObject) {
       event.setExtra("thread_name", loggingEvent.getThreadName());
     }
 
+    // remove keys with null values, there is no sense to send these keys to Sentry
     final Map<String, String> mdcProperties =
-        CollectionUtils.shallowCopy(loggingEvent.getMDCPropertyMap());
+        loggingEvent.getMDCPropertyMap().entrySet().stream()
```

Comment:
mind doing the same fix for log4j2 SentryAdapter? it's pretty much the same thing.